src/librustc/infer/region_inference/mod.rs

```diff
@@ -1543,7 +1543,7 @@ impl<'a, 'gcx, 'tcx> GenericKind<'tcx> {
         match *self {
             GenericKind::Param(ref p) => p.to_ty(tcx),
             GenericKind::Projection(ref p) => tcx.mk_projection(
-                p.trait_ref.clone(), p.item_name(tcx)),
+                p.trait_ref(tcx).clone(), p.item_name(tcx)),
```

Comment:
I assume you are suggesting that `mk_projection` should change its signature, and in particular likely take `item_def_id` directly? Before I go wild on this, what would the final signature be? There would be some fallout, some call sites have exactly `trait_ref` and `item_name` available.